pkgs/by-name/ni/niri/package.nix

```diff
@@ -20,19 +20,20 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "niri";
-  version = "0.1.7";
+  version = "0.1.8";
 
   src = fetchFromGitHub {
     owner = "YaLTeR";
     repo = "niri";
-    rev = "v${version}";
-    hash = "sha256-EdlOGL9RdO77HnZxv2UpPwrJdFH8qPrPSRtkBBV167s=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-13xynDWoOqogUKZTf6lz267hEQGdCE+BE6acs2G3j8k=";
   };
 
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = "${src}/Cargo.lock";
```

Comment:
@IogaMaster thoughts? Isn't the reason we put `Cargo.lock` in nixpkgs to prevent IFD or something? (is that at all right?). see prior discussion:
- https://github.com/NixOS/nixpkgs/pull/289578#issuecomment-1951416865
- https://github.com/NixOS/nixpkgs/pull/289578#issuecomment-1951418464

Reply:
Yeah the Cargo.lock generated and stored in nixpkgs is different iirc. So let's keep it as `lockFile = ./Cargo.lock; `

Reply:
I checked this on version 0.1.7 and they are no different.

Reply:
> Yeah the Cargo.lock generated and stored in nixpkgs is different iirc

No? They must be bitwise identical, as far as I'm aware. Even if identical TOML documents, builds still fail if the actual files aren't identical.

Reply:
Now I've checked, and no one in nixpkgs seems to be using "${src}/Cargo.lock", instead putting Cargo.lock in the package directory, so maybe I really rushed the replacement? Although I was able to build several packages in nixpkgs without errors by replacing lockFile with "${src}/Cargo.lock".

Reply:
the current implementation won't work as [ifd is disallowed in nixpkgs](https://github.com/NixOS/nixpkgs/tree/master/pkgs#import-from-derivation), and depending on a file from the source repository would add a dependency on output of `fetchFromGitHub` derivation above (as `buildRustCrate` reads this file from nix)

Reply:
> the proposed implementation won't

That's not true. It will work, and as I said, I've tested several packages, including this one, replacing everything with "Cargo.lock". But since it does look like an IFD, I agree that it's worth putting everything back in place.